File: mock1-uniqueIndex.js
function uniqueIndex(s) {
    var hashmap = {};
    for (var i = 0; i < s.length; i++){
        if (s[i] in hashmap) {
            hashmap[s[i]] += 1;
        }
        else {
            hashmap[s[i]] = 1;
        }
    }
    for (var i = 0; i < s.length; i++){
        if (hashmap[s[i]] == 1) {
            return i;
        }
    }
    return -1
}

console.log(uniqueIndex('leetcode'));

File: mock1-UniqueIndex.py
def uniqueIndex(s):
    d={}
    for i in s:
        if i not in d:
            d[i]=1
        else:
            d[i]+=1
    for i in range(len(s)):
        if d[s[i]]==1:
            return i
    return -1
print(uniqueIndex('leetcode'))
print(uniqueIndex('aabb'))
